src/index.js

```diff
@@ -84,51 +60,20 @@ loader.pitch = function(request) {
 	}
 
 	(new SingleEntryPlugin(this.context, `!!${path.resolve(__dirname, 'rpc-worker-loader.js')}!${request}`, 'main')).apply(worker.compiler);
-
-	const subCache = `subcache ${__dirname} ${request}`;
-
-	compilationHook(worker.compiler, (compilation, data) => {
-		if (compilation.cache) {
-			if (!compilation.cache[subCache]) compilation.cache[subCache] = {};
-
-			compilation.cache = compilation.cache[subCache];
-		}
-		parseHook(data, (parser, options) => {
-			exportDeclarationHook(parser, expr => {
-				let decl = expr.declaration || expr,
-					{ compilation, current } = parser.state,
-					entry = compilation.entries[0].resource;
-
-				// only process entry exports
-				if (current.resource!==entry) return;
-
-				let key = current.nameForCondition();
-				let exports = CACHE[key] || (CACHE[key] = {});
-
-				if (decl.id) {
-					exports[decl.id.name] = true;
-				}
-				else if (decl.declarations) {
-					for (let i=0; i<decl.declarations.length; i++) {
-						exports[decl.declarations[i].id.name] = true;
-					}
-				}
-				else {
-					console.warn('[workerize] unknown export declaration: ', expr);
-				}
-			});
-		});
-	});
-
 	worker.compiler.runAsChild((err, entries, compilation) => {
 		if (err) return cb(err);
 
 		if (entries[0]) {
-			worker.file = entries[0].files[0];
+			worker.file = Array.from(entries[0].files)[0];
 
-			let key = entries[0].entryModule.nameForCondition();
 			let contents = compilation.assets[worker.file].source();
-			let exports = Object.keys(CACHE[key] || {});
+
+			if (entries[0].entryModule.buildMeta.providedExports === true) {
+				// Can also occur if doing `export * from 'common js module'`
+				throw new Error('Attempted to load a worker implemented in CommonJS');
+			}
+
+			let exports = entries[0].entryModule.buildMeta.providedExports;
```

Comment:
Note that as per https://github.com/webpack/webpack/pull/8576 (landed on Webpack 5.0),  `buildMeta.providedExports` is no longer there

To retrieve the provided exports like in webpack 4, we would need to call `compilation.moduleGraph.getProvidedExports`, i.e. `let providedExports = compilation.moduleGraph.getProvidedExports(entries[0].entryModule);`